### ISSUE TITLE
Public form does not work

### ISSUE THREAD

**Issue by aleadco** [nextcloud/forms#462]
### Steps to reproduce
A public access form is created that will be shared by the link generated to users without access to the nextcloud service

### Expected behaviour
The expectation is that this link simply works in the same way that you share any file publicly, that is, you do not need to be authenticated or have an access account in the system to access.

### Actual behaviour
The server was unable to complete your request.

If it happens again, please send the technical details below to the server administrator.

The same error is always obtained but only with public links. If it is created for internal users there is no problem.

```
More details can be seen in the server log.
Technical details

     Remote address: xxx.xxx.xxx.xxx
     Request ID: mbPMnGhDbxLRIsT1Yv8S
```

## Server configuration detail

**Operating system:** Linux 4.19.0-9-amd64 #1 SMP Debian 4.19.118-2 (2020-04-29) x86_64

**Webserver:** nginx/1.19.0 (fpm-fcgi)

**Database:** mysql 10.5.3

**PHP version:** 

7.3.19-1+0~20200612.60+debian10~1.gbp6c8fe1
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, sodium, session, standard, cgi-fcgi, mysqlnd, PDO, xml, apcu, bz2, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, igbinary, imagick, intl, json, ldap, exif, mysqli, pdo_mysql, apc, posix, readline, redis, shmop, SimpleXML, smbclient, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Phar, libsmbclient, Zend OPcache

**Nextcloud version:** 18.0.6 - 18.0.6.0

**Updated from an older Nextcloud** 

`Error | index | Exception: Argument 2 passed to OCA\Forms\Db\NotificationMapper::findByUserAndForm() must be of the type string, null given, called in /var/www/nextcloud/apps/forms/lib/Controller/PageController.php on line 277/var/www/nextcloud/lib/private/AppFramework/App.php - line 125:OC\AppFramework\Http\Dispatcher->dispatch(OCA\Forms\Co ... {}, "gotoForm")/var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47:OC\AppFramework\App::main("OCA\\Forms\ ... r", "gotoForm", OC\AppFramew ... {}, { hash: "vVo ... "})<<closure>>OC\AppFramework\Routing\RouteActionHandler->__invoke({ hash: "vVo ... "})/var/www/nextcloud/lib/private/Route/Router.php - line 299:call_user_func(OC\AppFramew ... {}, { hash: "vVo ... "})/var/www/nextcloud/lib/base.php - line 1008:OC\Route\Router->match("/apps/forms ... q")/var/www/nextcloud/index.php - line 38:OC::handleRequest()Causado porTypeError: Argument 2 passed to OCA\Forms\Db\NotificationMapper::findByUserAndForm() must be of the type string, null given, called in /var/www/nextcloud/apps/forms/lib/Controller/PageController.php on line 277/var/www/nextcloud/apps/forms/lib/Controller/PageController.php - line 277:OCA\Forms\Db\NotificationMapper->findByUserAndForm(5, null)/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 170:OCA\Forms\Controller\PageController->gotoForm("vVorQvIos0AiTslq")/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 99:OC\AppFramework\Http\Dispatcher->executeController(OCA\Forms\Co ... {}, "gotoForm")/var/www/nextcloud/lib/private/AppFramework/App.php - line 125:OC\AppFramework\Http\Dispatcher->dispatch(OCA\Forms\Co ... {}, "gotoForm")/var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47:OC\AppFramework\App::main("OCA\\Forms\ ... r", "gotoForm", OC\AppFramew ... {}, { hash: "vVo ... "})<<closure>>OC\AppFramework\Routing\RouteActionHandler->__invoke({ hash: "vVo ... "})/var/www/nextcloud/lib/private/Route/Router.php - line 299:call_user_func(OC\AppFramew ... {}, { hash: "vVo ... "})/var/www/nextcloud/lib/base.php - line 1008:OC\Route\Router->match("/apps/forms ... q")/var/www/nextcloud/index.php - line 38:OC::handleRequest()
-- | -- | --`

**Comment by skjnldsv** [comment]
Hey  @aleadco 
What version of forms are you using?

**Comment by Kuphi** [comment]
Can confirm this:
**Internal Server Error**
The server was unable to complete your request.
If this happens again, please send the technical details below to the server administrator.
More details can be found in the server log.
Technical details

    Remote Address: xx.xxx.xx.xx
    Request ID: Wh7dqNHuVx7342Vb5yAv

Using Nextcloud 18.0.6 with Apache.

But I did the simple install in the web interface. Is it necessary to perform the command npm [...] for correct install?
https://github.com/nextcloud/forms/issues/465

**Comment by aleadco** [comment]
> What version of forms are you using?

Hi @skjnldsv  I have implemented the stable version 1.1.1 that I installed from the nextcloud web interface. In relation to #463  I can say that neither of the two variants works on public links, that is, neither with an expiration date nor with no established date.

**Comment by jotoeri** [comment]
Hey Guys,

this bug has been tackled already and will be solved with the upcoming release. The corresponding issue is pinned on top of the issues. You can also find possible quick-fix solutions there. => #405
Please next time have a look onto & search in old issues, before creating a new one. Just makes it easier for us, to support you. ;)

Closing this as duplicate.

Greets,
Jonas
